src/caa-cases/containers/caa-cases/caa-cases.component.ts

```diff
@@ -261,9 +266,57 @@ export class CaaCasesComponent implements OnInit {
         ? CaaCasesFilterType.CaseReferenceNumber
         : CaaCasesFilterType.None;
     }
+    if (this.selectedFilterType === CaaCasesFilterType.None) {
+      // Remove unassigned cases related filter from session
+      this.removeSessionState(this.caaCasesPageType);
+    } else {
+      // Store filter values to session
+      this.storeSessionState();
+    }
+    // Load case types based on current page type, selected filter type and selected filter value
     this.loadCaseTypes(this.selectedFilterType, this.selectedFilterValue);
   }
 
+  public removeSessionState(key: string): void {
+    this.service.removeSessionState(key);
+  }
+
+  public retrieveSessionState(): void {
+    this.sessionStateValue = this.service.retrieveSessionState(this.caaCasesPageType);
+    if (this.sessionStateValue) {
+      this.selectedFilterType = this.sessionStateValue.filterType ? this.sessionStateValue.filterType : null;
+      if (this.selectedFilterType) {
+        const caseReferenceNumber = this.sessionStateValue.caseReferenceNumber && this.sessionStateValue.caseReferenceNumber;
+        const assigneeName = this.sessionStateValue.assigneeName && this.sessionStateValue.assigneeName;
```

Comment:
@johnbenjamin-hmcts - duplicate `this.sessionStateValue.assigneeName` check

Reply:
@Daniel-Lam - I used this statement to get the `assignee name` only if it is defined.